automation/taskcluster/update_android_components.sh

```diff
@@ -5,52 +5,60 @@
 # If a command fails then do not proceed and fail this script too.
 set -ex
 
-# Install dependencies (TODO: Move to Docker image)
-apt-get install -y brew
-brew install hub
+export BRANCH="ac-update"
+export GITHUB_USER="MickeyMoz"
+export EMAIL="sebastian@mozilla.com"
+export REPO="reference-browser"
 
-BRANCH="ac-update"
-USER="MickeyMoz"
-EMAIL="sebastian@mozilla.com"
-REPO="reference-browser"
+git config --global user.email "$EMAIL"
+git config --global user.name "$GITHUB_USER"
 
-git config --global user.email $EMAIL
-git config --global user.name $USER
+COMPONENT_TO_WATCH='browser-engine-gecko-nightly'
+MAVEN_URL="https://nightly.maven.mozilla.org/maven2/org/mozilla/components/$COMPONENT_TO_WATCH"
 
-# Fetching latest version
-LATEST_VERSION=`curl https://nightly.maven.mozilla.org/maven2/org/mozilla/components/browser-engine-gecko-nightly/maven-metadata.xml | sed -ne '/latest/{s/.*<latest>\(.*\)<\/latest>.*/\1/p;q;}'`
+# Fetch latest version
+LATEST_VERSION=$(curl "$MAVEN_URL/maven-metadata.xml" | sed -ne '/latest/{s/.*<latest>\(.*\)<\/latest>.*/\1/p;q;}')
+
+# Check the latest version was uploaded by Mozilla
+LATEST_POM_URL="$MAVEN_URL/$LATEST_VERSION/$COMPONENT_TO_WATCH-$LATEST_VERSION.pom"
+POM_FILE='component.pom'
+$CURL "$LATEST_POM_URL" --output "$POM_FILE"
+$CURL "$LATEST_POM_URL.asc" --output "$POM_FILE.asc"
+gpg --verify "$POM_FILE.asc" "$POM_FILE"
 
 # Updating version file
 sed -i "s/VERSION = \".*\"/VERSION = \"$LATEST_VERSION\"/g" "buildSrc/src/main/java/AndroidComponents.kt"
 
 # Create a branch and commit local changes
-git checkout -b $BRANCH
+git checkout -b "$BRANCH"
 git add buildSrc/src/main/java/AndroidComponents.kt
 git commit -m \
 	"Update Android Components version to $LATEST_VERSION." \
 	--author="MickeyMoz <sebastian@mozilla.com>" \
 || { echo "No new Android Components version ($LATEST_VERSTION) available"; exit 0; }
 
-# Get token for using GitHub
-python automation/taskcluster/helper/get-secret.py \
-    -s project/mobile/github \
-    -k botAccountToken \
-    -f .github_token \
 
 # From here on we do not want to print the commands since they contain tokens
 set +x
 
-GITHUB_TOKEN=`cat .github_token`
-URL="https://$USER:$GITHUB_TOKEN@github.com/$USER/$REPO/"
+export GITHUB_TOKEN=$(cat .github_token)
+
+if [[ $GITHUB_TOKEN == 'faketoken' ]]; then
+    echo '"faketoken" detected, not pushing anything'
+    exit 0
+fi
 
 # Push changes to GitHub
 echo "Pushing branch to GitHub"
-git push -f --no-verify --quiet $URL $BRANCH > /dev/null 2>&1 || echo "Failed ($?)"
+URL="https://${GITHUB_USER}:${GITHUB_TOKEN}@github.com/$GITHUB_USER/$REPO/"
+git push --force --no-verify --quiet "$URL" "$BRANCH"
```

Comment:
In the previous version I piped all (error) output to `/dev/null`. In the past I saw `git` leak the URL including the token when the network request failed (regardless of `--quiet`).

Reply:
Ah, thank you very much for the context! I put your logic back 😃